### PR TITLE
[T4A3][W11-B3] Fan Weiguang

### PR DESCRIPTION
**Pull request by fanwgwg**
Ready for review

### CODE REVIEW

File: src/seedu/addressbook/data/person/Contact.java [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+import seedu.addressbook.data.exception.IllegalValueException;
+
+/**
+ * Represents the contact info of a person
+ *
+ */
+public class Contact {
+
+	public final String value;
```
Comment (jeffryhartanto) [inline src/seedu/addressbook/data/person/Contact.java]: This should be declared private for better encapsulation.

File: src/seedu/addressbook/data/person/Contact.java [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+	protected boolean isPrivate;
```
Comment (jeffryhartanto) [inline src/seedu/addressbook/data/person/Contact.java]: Why protected? should be declared as `private`.

File: src/seedu/addressbook/data/person/Contact.java [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+	public Contact(String value, boolean isPrivate, String example, String messageConstraints, String validationRegex)
+			throws IllegalValueException {
+		this.value = value;
+		this.isPrivate = isPrivate;
+		if (!value.matches(validationRegex)) {
+			throw new IllegalValueException(messageConstraints);
```
Comment (jeffryhartanto) [inline src/seedu/addressbook/data/person/Contact.java]: Good job for passing the `validationRegex` here.

File: src/seedu/addressbook/data/person/Contact.java [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+		return value;
+	}
+
+	@Override
+	public int hashCode() {
+		return value.hashCode();
+	}
+
+	public boolean isPrivate() {
+		return isPrivate;
```
Comment (jeffryhartanto) [inline src/seedu/addressbook/data/person/Contact.java]: Please fix the indentation and change the tab to spaces.

File: src/seedu/addressbook/data/person/Phone.java [hunk 5]
```diff
@@ -3,57 +3,30 @@
...
+	 */
+	public Phone(String phone, boolean isPrivate) throws IllegalValueException {
+		super(phone.trim(), isPrivate, EXAMPLE, MESSAGE_PHONE_CONSTRAINTS, PHONE_VALIDATION_REGEX);
+	}
+
+	@Override
+	public boolean equals(Object other) {
+		return other == this // short circuit if same object
+				|| (other instanceof Phone // instanceof handles nulls
+						&& this.value.equals(((Phone) other).value)); // state
```
Comment (jeffryhartanto) [inline src/seedu/addressbook/data/person/Phone.java]: Please fix the indentation and change the tab to spaces.

**Comment by jeffryhartanto** [comment]
Some comments added. Please close the PR after reading comments.

**Comment by fanwgwg** [comment]
Comments reviewed. The isPrivate value should be private, and I will fix the indentation format. Thank you very much!

**Pull request closed without merging**